### PR TITLE
Added enableFeedback property to FloatingActionButton

### PR DESCRIPTION
**Pull request by chinmoy12c**
## Description

Added enableFeedback property to FloatingActionButton and FloatingActionButtonThemeData.

## Related Issues

#69397 

## Tests

I added the following tests:

- FloatingActionButton with enabled feedback.
- FloatingActionButton with disabled feedback.
- FloatingActionButton with enabled feedback by default.
- FloatingActionButton with disabled feedback using BottomNavigationBarTheme.
- FloatingActionButton.enableFeedback overrides FloatingActionButtonThemeData.enableFeedback

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by chinmoy12c** [comment]
Hey @Piinks, can you please review this one. It's related to the enableFeedback issue of standard widgets.

File: packages/flutter/lib/src/material/floating_action_button.dart [hunk 1]
```diff
@@ -416,6 +418,16 @@ class FloatingActionButton extends StatelessWidget {
   ///  * [MaterialTapTargetSize], for a description of how this affects tap targets.
   final MaterialTapTargetSize? materialTapTargetSize;
 
+  /// Whether detected gestures should provide acoustic and/or haptic feedback.
+  ///
+  /// For example, on Android a tap will produce a clicking sound and a
+  /// long-press will produce a short vibration, when feedback is enabled.
+  ///
```
Comment (Piinks) [inline packages/flutter/lib/src/material/floating_action_button.dart:425]: We should add mention here that when null, the theme will be used, and if that is null what the default behavior is.

**Review by Piinks — APPROVED** [review]
LGTM with dic nit. Thanks for the contribution!

File: packages/flutter/lib/src/material/floating_action_button.dart (hunk 1 above)
Comment (chinmoy12c) [reply packages/flutter/lib/src/material/floating_action_button.dart:425]: Sure I'll update the docs.

Comment (chinmoy12c) [reply packages/flutter/lib/src/material/floating_action_button.dart:418]: @Piinks, its done :)

**Comment by chinmoy12c** [comment]
Hey @Piinks, can you please merge the PR 😄

**Pull request merged**